packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -265,21 +265,101 @@ function makePreparableAnalyticsAction<
   return rootAction as PreparableAnalyticsAction<EventType, StateNeeded>;
 }
 
-export const makeAnalyticsAction = <
+export type AnalyticsActionOptions<
   EventType extends AnalyticsType,
   StateNeeded extends
     StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
->(
-  prefix: string,
-  analyticsType: EventType,
-  getBuilder: (
+> = LegacyAnalyticsOptions<EventType, StateNeeded> & NextLegacyOptions;
+
+export interface NextLegacyOptions {
+  todo?: never; //Remove when we have at least one property in this interface
+}
+export interface LegacyAnalyticsOptions<
+  EventType extends AnalyticsType,
+  StateNeeded extends
+    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
+> {
+  prefix: string;
+  analyticsType: EventType;
+  __legacy__getBuilder: (
     client: CoveoSearchPageClient,
     state: StateNeeded
-  ) => Promise<EventBuilder | null> | null,
-  provider: (getState: () => StateNeeded) => SearchPageClientProvider = (
-    getState
-  ) => new SearchAnalyticsProvider(getState)
-): PreparableAnalyticsAction<WrappedAnalyticsType<EventType>, StateNeeded> => {
+  ) => Promise<EventBuilder | null> | null;
+  __legacy__provider?: (
+    getState: () => StateNeeded
+  ) => SearchPageClientProvider;
+}
+
+export function makeAnalyticsAction<
+  EventType extends AnalyticsType,
+  StateNeeded extends
+    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
+>(
+  prefix: string,
+  analyticsType: LegacyAnalyticsOptions<
```

Comment:
For some small readability gains, can we have named type for all this ?
For example:
```typescript
type __legacy__getBuilder = LegacyAnalyticsOptions<
    EventType,
    StateNeeded
  >['__legacy__getBuilder']

[...]


__legacy__get_builder: __legacy__getBuilder,
__legacy__provider: __legacy__provider,
```

Reply:
a5e9cd9a53cb3a546fbdfae6bea37e7799a4eedf